src/main/java/jdbchomework/dao/jdbc/AbstractDao.java

```diff
@@ -2,118 +2,116 @@
-        } catch (HibernateException e) {
-            if (transaction != null) {
-                transaction.rollback();
+        try (PreparedStatement statement = connection.prepareStatement("INSERT  INTO " + table + " (name)VALUES (?);")) {
+            connection.setTransactionIsolation(Connection.TRANSACTION_READ_COMMITTED);
+            connection.setAutoCommit(false);
+            statement.setString(1, toAdd.getName());
+            connection.commit();
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            log.error("Cannot connect to DB", e);
+            throw new RuntimeException("Cannot connect to DB", e);
+        } finally {
+            try {
+                connection.setAutoCommit(true);
+            } catch (SQLException e) {
+                log.error("Cannot connect to DB", e);
+                throw new RuntimeException("Cannot connect to DB", e);
             }
-            e.printStackTrace();
         }
     }
 
     public List<T> getAll() {
         List<T> result = new ArrayList<>();
-        Session session = getSession();
-        Transaction transaction = null;
-        try {
-            transaction = session.beginTransaction();
-            result = session.createQuery("FROM " + entity).list();
-            transaction.commit();
-        } catch (HibernateException e) {
-            if (transaction != null) {
-                transaction.rollback();
+        try (Statement statement = connection.createStatement()) {
+            String sql = "SELECT * FROM " + table;
+            ResultSet resultSet = statement.executeQuery(sql);
+            while (resultSet.next()) {
+                T t = createEntity(resultSet);
+                result.add(t);
             }
-            log.error(CANNOT_CONNECT_TO_DB, e);
+        } catch (SQLException e) {
+            log.error("Cannot connect to DB", e);
+            throw new RuntimeException("Cannot connect to DB", e);
         }
         return result;
     }
 
 
     public T getById(long id) {
         T result = null;
-        Session session = getSession();
-        Transaction transaction = null;
-        try {
-            transaction = session.beginTransaction();
-            result = session.get(clazz, id);
-            transaction.commit();
-        } catch (HibernateException e) {
-            if (transaction != null) {
-                transaction.rollback();
+        try (
+                PreparedStatement statement = connection
+                        .prepareStatement("SELECT * FROM " + table + " WHERE " + column + " = ?;")) {
+            statement.setLong(1, id);
+            ResultSet rs = statement.executeQuery();
+            if (rs.next()) {
+                result = createEntity(rs);
             }
-            log.error(CANNOT_CONNECT_TO_DB, e);
+        } catch (SQLException e) {
+            log.error("Cannot connect to DB", e);
+            throw new RuntimeException("Cannot connect to DB", e);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)